apollo-gradle-plugin-incubating/src/main/kotlin/com/apollographql/apollo/gradle/internal/JvmTaskConfigurator.kt

```diff
@@ -10,33 +10,42 @@ object JvmTaskConfigurator {
   fun getVariants(project: Project): NamedDomainObjectContainer<ApolloVariant> {
     val container = project.container(ApolloVariant::class.java)
 
-    // TODO: should we add tasks for the test sourceSet ?
-    val name = "main"
-    val apolloVariant = ApolloVariant(
-        name = name,
-        sourceSetNames = listOf(name),
-        androidVariant = null
-    )
-
-    container.add(apolloVariant)
+    val javaPlugin = project.convention.getPlugin(JavaPluginConvention::class.java)
+    val sourceSets = javaPlugin.sourceSets
+
+    sourceSets.map { it.name }.forEach {name ->
+      val apolloVariant = ApolloVariant(
+          name = name,
+          sourceSetNames = listOf(name),
+          androidVariant = null
+      )
+
+      container.add(apolloVariant)
+    }
+
     return container
   }
 
   fun registerGeneratedDirectory(project: Project, compilationUnit: DefaultCompilationUnit, codegenProvider: TaskProvider<ApolloGenerateSourcesTask>) {
     val javaPlugin = project.convention.getPlugin(JavaPluginConvention::class.java)
     val sourceSets = javaPlugin.sourceSets
-    val name = compilationUnit.variantName
+    val sourceSetName = compilationUnit.variantName
+
+    val taskName = when(compilationUnit.variantName) {
+      "main" -> ""
```

Comment:
Just double checking. Main name is omit only when there is a single source set. If there are multiple, main is used explicitly. I.e compileMainKotlin

Reply:
Thanks for the heads up :+1: , I'll add a test and patch this.

Reply:
I'm 90% sure but please verify this with tests beforehand :)

Reply:
Looks like it's still called `compileJava`/`compileKotlin`... I'll try to make a small sample repo

Reply:
I added the check to one of the tests: https://github.com/apollographql/apollo-android/pull/1764/files#diff-b308f8b88ded6b4cfed0ca665d8de6f8R161